lib/deaf_grandma.rb

```diff
@@ -19,13 +19,33 @@ def run!
 
 
   def speak(input)
+    #Check for yel
+
+    if input == input.upcase
+      if input == "BYE"
+          bye_function
+      else
+        "NOT SINCE 1964!"
+      end
+    else
+      "SPEAK UP SONNY!"
+    end
 
-    #Implement your code here <<<<<<<<<
 
   end
 
   private
 
+  def bye_function
+    if @bye_counter < 2
+      @bye_counter += 1
+      ""
+    else
+        @bye_counter = 0
```

Comment:
You don't need to reset the bye counter here and see if you can get the program to exit after the 3rd bye.